packages/pyright-internal/src/tests/typeEvaluatorBased.test.ts

```diff
@@ -156,3 +156,20 @@ test('`reportUnusedFunction` on `@final` classes', () => {
         ],
     });
 });
+
+describe('uninitialized variable checking with init method calling', () => {
```

Comment:
another thing to consider is that it also doesn't work when the initialization is nested in a second method call:

```py
class C:
    x: int # error

    def __init__(self) -> None:
        self.reset()
    
    def reset(self) -> None:
        self.reset2()
    
    def reset2(self) -> None:
        self.x = 1
```

though this probably more of an edge case and would probably destroy performance since it would need to recursively check every method call in the class, so we probably shouldn't do this.

i think we should document this though (a new file in https://github.com/DetachHead/basedpyright/tree/main/docs/benefits-over-pyright probably). im happy to do this myself if you'd prefer

Reply:
Yeah, you're welcome to work on documenting it.